cmake/onnxruntime_providers_qnn.cmake

```diff
@@ -3,41 +3,90 @@
 
   add_compile_definitions(USE_QNN=1)
 
-  # These are shared utils,
-  # TODO, move to a separate lib when used by EPs other than QNN, NNAPI and CoreML
-  file(GLOB onnxruntime_providers_shared_utils_cc_srcs CONFIGURE_DEPENDS
-    "${ONNXRUNTIME_ROOT}/core/providers/shared/utils/utils.h"
-    "${ONNXRUNTIME_ROOT}/core/providers/shared/utils/utils.cc"
-  )
+  if(onnxruntime_BUILD_QNN_EP_STATIC_LIB)
+    add_compile_definitions(BUILD_QNN_EP_STATIC_LIB=1)
+  endif()
 
   file(GLOB_RECURSE
-    onnxruntime_providers_qnn_ep_cc_srcs CONFIGURE_DEPENDS
-    "${ONNXRUNTIME_ROOT}/core/providers/qnn/*.h"
-    "${ONNXRUNTIME_ROOT}/core/providers/qnn/*.cc"
+       onnxruntime_providers_qnn_ep_srcs CONFIGURE_DEPENDS
+       "${ONNXRUNTIME_ROOT}/core/providers/qnn/*.h"
+       "${ONNXRUNTIME_ROOT}/core/providers/qnn/*.cc"
   )
 
-  file(GLOB_RECURSE
-    onnxruntime_providers_qnn_builder_cc_srcs CONFIGURE_DEPENDS
-    "${ONNXRUNTIME_ROOT}/core/providers/qnn/builder/*.h"
-    "${ONNXRUNTIME_ROOT}/core/providers/qnn/builder/*.cc"
-  )
+  if(onnxruntime_BUILD_QNN_EP_STATIC_LIB)
+    #
+    # Build QNN EP as a static library
+    #
+    set(onnxruntime_providers_qnn_srcs ${onnxruntime_providers_qnn_ep_srcs})
+    source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_qnn_srcs})
+    onnxruntime_add_static_library(onnxruntime_providers_qnn ${onnxruntime_providers_qnn_srcs})
+    onnxruntime_add_include_to_target(onnxruntime_providers_qnn onnxruntime_common onnxruntime_framework onnx
+                                                                onnx_proto protobuf::libprotobuf-lite
+                                                                flatbuffers::flatbuffers Boost::mp11)
+    target_link_libraries(onnxruntime_providers_qnn)
```

Comment:
This line is empty

Reply:
thanks. I removed this line (it is currently there in main branch for building qnn ep as static lib).